build.go

```diff
@@ -137,6 +147,39 @@ func Build(
 		logger.Action("Completed in %s", duration.Round(time.Millisecond))
 		logger.Break()
 
+		solver := GetEnvOrDefault("BP_CONDA_SOLVER", "conda")
+
+		if solver == "mamba" {
+			logger.Subprocess("Installing mamba solver")
+
+			conda := pexec.NewExecutable(condaLayer.Path + "/bin/conda")
+			duration, err = clock.Measure(func() error {
+				return conda.Execute(pexec.Execution{
+					Args: []string{"install", "-n", "base", "conda-libmamba-solver", "-y"},
```

Comment:
Reading through https://github.com/orgs/paketo-buildpacks/discussions/231 - I am inclined towards the path of having a miniforge buildpack. See https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html#existing-conda-install-not-recommended - this way of installation is not recommended by the upstream maintainers. Given that Paketo buildpacks strive to set best-practices, I'm not really in favor of this.

Ping @paketo-buildpacks/python-maintainers for their thoughts.
My recommendation would be for you to propose an RFC with how this workflow can be implemented with an independent miniforge buildpack and its own order group. You'll need to do some research on if the `conda-env-update` buildpack can handle both miniconda and miniforge.

Reply:
### current proposition
This patch does not install mamba on top of conda. It follows the suggestion from this [anaconda article](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community) where only the solver is installed and switched over to so there's nothing mamba added.

---
### mamba on top of conda
To be complete, I tested the mamba on top of conda installation variant to see what the effects were and basically, things break during the environment creation phase so beside being strongly not recommended, it is also not working in this buildpack context.

---
### miniforge variant 
For `conda-env-update`, one possible way to do things simple would be to have an alias or a symbolic link from `mamba` to `conda` created by the `miniforge` buildpack and thus no change required to `conda-env-update`. Otherwise, I currently don't know if there's a way for a previous buildpack to set information that can be reused by a later buildpack. One thing that I tested was to use an environment variable that would contain the command to execute and modify slightly `conda-env-update` so that the command object creation uses it in place of the current hard coded conda command.

---
### python-start
One other package that also might need some tweaks is [python-start](https://github.com/paketo-buildpacks/python-start/) where we would need to allow people to switch between conda and miniforge as the file based detection does not really allow to know which one should be used.
I can currently image two possible ways:
- an environment variable
- an configuration file to be put in the project a bit like the one used to configure which version of python should be installed